frontend/src/app/modules/global_search/global-search-input.component.ts

```diff
@@ -199,54 +121,59 @@ export class GlobalSearchInputComponent implements OnDestroy {
     event.stopPropagation();
     event.preventDefault();
 
+    // handle click on search button
     if (ContainHelpers.insideOrSelf(this.btn.nativeElement, event.target)) {
       this.submitNonEmptySearch();
     }
   }
 
-  public redirectToWp(id:string) {
-    window.location = this.PathHelperService.workPackagePath(id) as unknown as Location;
-  }
+  // load selected item
+  public onChange($event:any) {
+    // update embeeded table and title when new search is submitted
+    this.globalSearchService.searchTerm = this.searchTerm;
 
-  private unregister() {
-    if (this.unregisterGlobalListener) {
-      this.unregisterGlobalListener();
-      this.unregisterGlobalListener = undefined;
+    let selectedOption = $event;
+    if (selectedOption.id) {  // item is a work package element
+      this.redirectToWp(selectedOption.id);
+    } else {                  // item is a 'scope' element
+      this.searchInScope(selectedOption.projectScope);
     }
   }
 
-  public submitNonEmptySearch(forcePageLoad:boolean = false) {
-    this.globalSearchService.searchTerm = this.searchValue;
-    if (this.searchValue !== '') {
-      // Work package results can update without page reload.
-      if (!forcePageLoad &&
-          this.globalSearchService.isAfterSearch() &&
-          this.globalSearchService.currentTab === 'work_packages') {
-        window.history
-          .replaceState({},
-            `${I18n.t('global_search.search')}: ${this.searchValue}`,
-            this.globalSearchService.searchPath());
+  // close menu when input field is empty
+  public toggleMenu(searchedTerm:string) {
+    (searchedTerm.trim().length > 0) ? this.ngSelectComponent.isOpen = true : this.ngSelectComponent.isOpen = false;
```

Comment:
I prefer "if then else" writing here as you assign a value within each case. Or you do something like this:

```suggestion
    this.ngSelect.isOpen = (searchedTerm.trim().length > 0);
```